### PR TITLE
CDAP-8229 Don't depend on default namespace in UpgradeTool.

### PR DESCRIPTION
**Pull request by anwar6953**
[CDAP-8229](https://issues.cask.co/browse/CDAP-8229) Don't depend on default namespace in UpgradeTool.

http://builds.cask.co/browse/CDAP-RUT514-1

### CODE REVIEW

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeDatasetServiceManager.java [hunk 1]
```diff
@@ -116,7 +118,7 @@ protected void startUp() throws Exception {
       @Override
       public Boolean call() throws Exception {
         try {
-          getDSFramework().getInstances(Id.Namespace.DEFAULT);
+          namespaceQueryAdmin.list();
```
Comment (chtyim) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeDatasetServiceManager.java]: That's a bit awkward to use namespace query admin to test whether dataservice is up or not. Can we just try to list the `SYSTEM` namespace?

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeDatasetServiceManager.java [hunk 2]
```diff
@@ -116,7 +116,7 @@ protected void startUp() throws Exception {
       @Override
       public Boolean call() throws Exception {
         try {
-          getDSFramework().getInstances(Id.Namespace.DEFAULT);
+          getDSFramework().getInstances(NamespaceId.SYSTEM.toId());
```
Comment (chtyim) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeDatasetServiceManager.java:119]: Isn't `DatasetFramework.getInstances` take `NamespaceId`?? Why need to call `toId()`??

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeDatasetServiceManager.java (hunk 1 above)
Comment (anwar6953) [reply cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeDatasetServiceManager.java]: Yea, that would work too.

File: cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeDatasetServiceManager.java (hunk 2 above)
Comment (anwar6953) [reply cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeDatasetServiceManager.java:119]: In release/3.5 it took the old class. For the patch on later versions, we won't need the `toId()`.

Comment (chtyim) [reply cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeDatasetServiceManager.java:119]: oh, nevermind it's 3.5. It's ok.

**Comment by chtyim** [comment]
LGTM

**Pull request merged**